### PR TITLE
[CMAKE] Fix host tools build on x86-64 mingw

### PR DESCRIPTION
**Pull request by zefklop**
Dynamically check for sys/types.h and pid_t in wine config.h
Use TARGET_xxx defines instead of _X86_ as this is undefined by GCC
Add some sense in include directories management by using interface
libraries.

### CODE REVIEW

File: sdk/include/host/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    set(CMAKE_EXTRA_INCLUDE_FILES "sys/types.h")
+    #this sets HAVE_PID_T
+    CHECK_TYPE_SIZE("pid_t" PID_T)
+    unset(CMAKE_EXTRA_INCLUDE_FILES)
+endif()
+
+configure_file(config.h.in config.h @ONLY)
+
+add_library(host_includes INTERFACE)
+target_include_directories(host_includes INTERFACE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (SergeGautherie) [inline sdk/include/host/CMakeLists.txt]: Missing EOL.

File: sdk/tools/fatten/CMakeLists.txt [hunk 2]
```diff
@@ -4,3 +4,4 @@ add_host_tool(fatten
     fatfs/diskio.c
     fatfs/ff.c
     fatfs/option/ccsbcs.c)
+target_link_libraries(fatten PRIVATE host_includes)
```
Comment (SergeGautherie) [inline sdk/tools/fatten/CMakeLists.txt]: Missing EOL.

**Comment by ColinFinck** [comment]
This PR went through quickly and lacks some important details. What was the actual build error you encountered? How is that fixed by sys/types.h and pid_t?

I like your cleanup of the host include management, but adding `CHECK_INCLUDE_FILE("sys/types.h" HAVE_SYS_TYPES_H)` for a header that is guaranteed to be available on all our host platforms sets a questionable precedent. I don't want anyone to think that we need further PRs to replace all our working `#define HAVE_XYZ` statements by dynamic checks, essentially turning our build system into a slow 80s-style `./configure` script checking for even the weirdest situations.

**Comment by zefklop** [comment]
Wine includes define their own pid_t. Msys2 provided x86_64 gcc (9.2.0) defines this also, so this conflicts.

I'm fine if we get rid of the check for sys/types.h though

**Comment by ColinFinck** [comment]
Ah, I see! Please check my follow-up PR: https://github.com/reactos/reactos/pull/2052

**Pull request merged**